File: CS/DXApplication1/Form1.cs
using System.Collections.Generic;
using DevExpress.XtraBars.Ribbon;
using DevExpress.Spreadsheet;

namespace DXApplication1 {
    public partial class Form1 : RibbonForm {

        nwindDataSet dataSet;
        nwindDataSetTableAdapters.CategoriesTableAdapter adapter;
        IWorkbook template;

        public Form1() {
            InitializeComponent();

            dataSet = new nwindDataSet();
            adapter = new nwindDataSetTableAdapters.CategoriesTableAdapter();
            adapter.Fill(dataSet.Categories);

            template = spreadsheetControl1.Document;

            template.MailMergeDataSource = dataSet;
            template.MailMergeDataMember = "Categories";
        }

        private void Form1_Load(object sender, System.EventArgs e) {
            spreadsheetControl1.LoadDocument("Documents\\MailMergeTemplate.xlsx");
        }

        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            IList<IWorkbook> resultWorkbooks = spreadsheetControl1.Document.GenerateMailMergeDocuments();

            string fileName;
            int index = 0;

            foreach (IWorkbook workbook in resultWorkbooks) {
                fileName = string.Format("Documents\\SavedDocument{0}" + ".xlsx", index++);
                workbook.SaveDocument(fileName, DocumentFormat.OpenXml);
            }
        }
    }
}

File: VB/DXApplication1/Form1.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.Spreadsheet

Partial Public Class Form1
    Inherits RibbonForm

    Private dataSet As nwindDataSet
    Private adapter As nwindDataSetTableAdapters.CategoriesTableAdapter
    Private template As IWorkbook

	Public Sub New()
        InitializeComponent()

        dataSet = New nwindDataSet()
        adapter = New nwindDataSetTableAdapters.CategoriesTableAdapter()
        adapter.Fill(dataSet.Categories)

        template = SpreadsheetControl1.Document

        template.MailMergeDataSource = dataSet
        template.MailMergeDataMember = "Categories"
	End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        SpreadsheetControl1.LoadDocument("Documents\MailMergeTemplate.xlsx")
    End Sub

    Private Sub BarButtonItem1_ItemClick(sender As Object, e As DevExpress.XtraBars.ItemClickEventArgs) Handles BarButtonItem1.ItemClick
        Dim resultWorkbooks As IList(Of IWorkbook) = SpreadsheetControl1.Document.GenerateMailMergeDocuments()

        Dim fileName As String
        Dim index As Integer = 0

        For Each workbook As IWorkbook In resultWorkbooks
            fileName = String.Format("C:\TEMP\SavedDocument{0}" & ".xlsx", index)
            index += 1
            workbook.SaveDocument(fileName, DocumentFormat.OpenXml)
        Next workbook
    End Sub
End Class
